AnalysisAlgos/SiStripClusterInfoProducer/plugins/SiStripProcessedRawDigiProducer.cc

```diff
@@ -31,16 +29,16 @@ void SiStripProcessedRawDigiProducer::produce(edm::Event& e, const edm::EventSet
   edm::Handle<edm::DetSetVector<SiStripDigi> > inputDigis;
   edm::Handle<edm::DetSetVector<SiStripRawDigi> > inputRawdigis;
 
-  es.get<SiStripGainRcd>().get(gainHandle);
-  subtractorPed->init(es);
-  subtractorCMN->init(es);
+  gainHandle_ = es.getHandle(gainToken_);
```

Comment:
Why do you also save  these Handle's? According to https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToGetDataFromES the data should be accessed directly with the token, while the handle should be reserved for cases in which "the caller want's to check whether the data product exists", which is not the case here.

Reply:
I missed a few of those indeed, I went through the list again and removed those without a check for the validity of the handle.